devito/types/basic.py

```diff
@@ -386,13 +386,24 @@ def _filter_assumptions(cls, **kwargs):
 
         return assumptions, kwargs
 
+    @staticmethod
+    def __xnew__(cls, name, **assumptions):
+        # Create the new Symbol
+        # Note: use __xnew__ to bypass sympy caching
+        newobj = sympy.Symbol.__xnew__(cls, name, **assumptions)
+
+        assumptions = newobj._assumptions.copy()
+        for key in ('real', 'imaginary', 'complex'):
+            assumptions.pop(key, None)
+        newobj._assumptions = assumptions
```

Comment:
aren't these four lines equivalent to just

```
newobj._assumptions = {k: v for k, v in newobj._assumptions.items()
                       if k not in ('real', 'imaginary', 'complex')}
```

Reply:
`_assumptions` is some fnacy sympy internal type not a plain dict so this would break a lot of stuff.